perllib/FixMyStreet/Cobrand/Peterborough.pm

```diff
@@ -1361,4 +1361,92 @@ sub bulky_available_feature_types {
     return { map { $_->{ID} => $_->{Name} } @types };
 }
 
+sub bulky_nice_collection_date {
+    my ($self, $date) = @_;
+    my $parser = DateTime::Format::Strptime->new( pattern => '%FT%T' );
+    my $dt = $parser->parse_datetime($date)->truncate( to => 'day' );
+    return $dt->strftime('%d %B');
+}
+
+sub bulky_nice_item_list {
+    my ($self, $report) = @_;
+
+    my @fields = sort grep { $_->{name} =~ /ITEM_/ } @{$report->get_extra_fields};
+    return [ map { $_->{value} } @fields ];
+}
+
+sub bulky_reminders {
+    my ($self, $params) = @_;
+
+    # Can't see an easy way to find these apart from loop through them all.
+    # Is only daily.
+    my $collections = FixMyStreet::DB->resultset('Problem')->search({
+        category => 'Bulky collection',
+        state => [ FixMyStreet::DB::Result::Problem->open_states ], # XXX?
+    });
+    my $parser = DateTime::Format::Strptime->new( pattern => '%FT%T' );
+    my $now = DateTime->now->set_time_zone(FixMyStreet->local_time_zone);
+
+    while (my $report = $collections->next) {
+        my $r1 = $report->get_extra_metadata('reminder_1');
+        my $r3 = $report->get_extra_metadata('reminder_3');
+        next if $r1; # No reminders left to do
+
+        # XXX Need to check if the subscription has been cancelled and if so,
+        # do not send out the email.
+
+        my $date = $report->get_extra_field_value('DATE');
+        my $dt = $parser->parse_datetime($date)->truncate( to => 'day' );
+        my $d1 = $dt->clone->subtract(days => 1);
+        my $d3 = $dt->clone->subtract(days => 3);
+
+        my $h = {
+            report => $report,
+            cobrand => $self,
+        };
+
+        if (!$r3 && $now >= $d3 && $now < $d1) {
+            $h->{days} = 3;
+            $self->_bulky_send_reminder_email($report, $h, $params);
+            $report->set_extra_metadata(reminder_3 => 1);
+            $report->update;
+        } elsif ($now >= $d1 && $now < $dt) {
+            $h->{days} = 1;
+            $self->_bulky_send_reminder_email($report, $h, $params);
+            $report->set_extra_metadata(reminder_1 => 1);
+            $report->update;
+        }
+    }
+}
+
+sub _bulky_send_reminder_email {
+    my ($self, $report, $h, $params) = @_;
+
+    my $token = FixMyStreet::App->model('DB::Token')->new({
```

Comment:
Think you need to `use FixMyStreet::App;` up top otherwise this line errors (at least it does here):

```
Can't locate object method "model" via package "FixMyStreet::App" at [...]/fixmystreet/perllib/FixMyStreet/Cobrand/Peterborough.pm line 1428.
```

Though adding that line then results in
```
Can't call method "get" on an undefined value at [...]/fixmystreet/commonlib/perllib/mySociety/Locale.pm line 22.
```
but not sure if that's my setup?

Reply:
Ugh, shouldn't have used App, it's not running in the web, it's offline. Guess it is there when tests are running. Should be FixMyStreet::DB.

Reply:
Aha, in that case you need `use FixMyStreet::Email;`, otherwise:

```
Undefined subroutine &FixMyStreet::Email::send_cron called at [...]/fixmystreet/perllib/FixMyStreet/Cobrand/Peterborough.pm line 1442.
```